google-datacatalog-rdbms-connector/tests/google/datacatalog_connectors/rdbms/sync/datacatalog_synchronizer_test.py

```diff
@@ -88,15 +94,75 @@ def test_synchronize_metadata_should_not_raise_error(  # noqa: E125
             enable_monitoring=True)
 
         synchronizer.run()
-        self.assertEqual(get_metadata.call_count, 1)
+        self.assertEqual(scrape.call_count, 1)
         self.assertEqual(
             synchronizer.
             _DataCatalogSynchronizer__metadata_definition['database_name'],
             'test_db')
-        self.assertEqual(make_entries_from_table_container_metadata.call_count,
-                         1)
+        self.assertEqual(make_base_entries.call_count, 1)
         self.assertEqual(ingest_metadata.call_count, 1)
         self.assertEqual(delete_obsolete_metadata.call_count, 1)
         self.assertEqual(process_entries_length_metric.call_count, 1)
         self.assertEqual(process_metadata_payload_bytes_metric.call_count, 1)
         self.assertEqual(process_elapsed_time_metric.call_count, 1)
+
+    @mock.patch('google.datacatalog_connectors.rdbms.'
+                'scrape.metadata_scraper.MetadataScraper.scrape')
+    @mock.patch('google.datacatalog_connectors.rdbms.'
+                'prepare.sql_objects.assembled_sql_objects_entry_factory.'
+                'AssembledSQLObjectsEntryFactory.make_entries')
+    @mock.patch('google.datacatalog_connectors.rdbms.'
+                'prepare.assembled_entry_factory.'
+                'AssembledEntryFactory.make_entries')
+    @mock.patch('google.datacatalog_connectors.commons.ingest.'
+                'datacatalog_metadata_ingestor.'
+                'DataCatalogMetadataIngestor.ingest_metadata')
+    @mock.patch('google.datacatalog_connectors.commons.cleanup.'
+                'datacatalog_metadata_cleaner.DataCatalogMetadataCleaner.'
+                'delete_obsolete_metadata')
+    @mock.patch('google.datacatalog_connectors.commons.monitoring.'
+                'metrics_processor.MetricsProcessor.'
+                'process_elapsed_time_metric')
+    @mock.patch('google.datacatalog_connectors.commons.monitoring.'
+                'metrics_processor.MetricsProcessor.'
+                'process_metadata_payload_bytes_metric')
+    @mock.patch('google.datacatalog_connectors.commons.monitoring.'
+                'metrics_processor.MetricsProcessor.'
+                'process_entries_length_metric')
+    def test_synchronize_metadata_with_sql_config_should_not_raise_error(  # noqa: E501
```

Comment:
So many mock objects are odd... Maybe it's time to think about refactoring the code as a follow-up to this PR. WDYT?

Reply:
makes sense, I will do this on the follow-up PR.